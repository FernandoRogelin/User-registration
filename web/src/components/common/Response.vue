<script setup>
defineProps({
  response: {
    type: Object,
    required: true
  }
})
</script>

<template>
  <div class="components-common-response" :class="{ 'components-common-response--error': response?.error }">
    <span class="components-common-response__message">
      {{ response?.message || response?.error }}
    </span>
  </div>
</template>

<style scoped lang="scss">
.components-common-response {
  left: 50%;
  top: 2rem;
  padding: 1rem;
  position: fixed;
  font-size: 0.8rem;
  border-radius: 4px;
  transform: translateX(-50%);
  background-color: var(--mb-white);
  border: 1px solid var(--mb-orange);
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  animation: slideUp 0.3s ease-out forwards;

  &--error {
    color: var(--mb-error);
    border-color: var(--mb-error);
  }

  &__message {
    display: block;
    text-align: center;
    font-weight: 500;
  }
}

@keyframes slideUp {
  from {
    opacity: 0;
    transform: translate(-50%, 1rem);
  }
  to {
    opacity: 1;
    transform: translate(-50%, 0);
  }
}
</style>
