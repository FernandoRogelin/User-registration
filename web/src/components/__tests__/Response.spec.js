import { describe, expect, test, afterEach } from "vitest"
import { render, screen, cleanup } from "@testing-library/vue"

import Response from "@/components/common/Response.vue"

const mockProps = { response: { message: 'Response component' } }

afterEach(() => cleanup())

describe('<Response /> common component', () => {
  test('should render component and match snapshot', () => {
    const { container } = render(Response, { props: mockProps })

    expect(container.firstChild).toMatchSnapshot()
  })

  test('should render message', () => {
    render(Response, { props: mockProps })

    expect(screen.getByText(/Response component/i))
  })
})
