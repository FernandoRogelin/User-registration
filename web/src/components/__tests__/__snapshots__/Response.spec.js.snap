// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Response /> common component > should render component and match snapshot 1`] = `
<div
  class="components-common-response"
  data-v-154c7870=""
>
  <span
    class="components-common-response__message"
    data-v-154c7870=""
  >
    Response component
  </span>
</div>
`;
